frontend/src/views/Lobby.tsx

```diff
@@ -1,31 +1,93 @@
 import React, { useCallback, useEffect, useState } from 'react';
 import { useLocation, useHistory } from 'react-router-dom';
 import { Message, Subscription } from 'stompjs';
+import styled from 'styled-components';
+import copy from 'copy-to-clipboard';
 import ErrorMessage from '../components/core/Error';
-import { LargeText, MediumText, Text } from '../components/core/Text';
+import {
+  NoMarginMediumText,
+  MainHeaderText,
+  SmallHeaderText,
+  NoMarginSubtitleText,
+} from '../components/core/Text';
 import {
   connect, routes, subscribe, disconnect,
 } from '../api/Socket';
 import { User } from '../api/User';
 import { checkLocationState, isValidRoomId } from '../util/Utility';
 import { Difficulty } from '../api/Difficulty';
-import { PrimaryButton, DifficultyButton } from '../components/core/Button';
+import { PrimaryButton, SmallDifficultyButton } from '../components/core/Button';
 import Loading from '../components/core/Loading';
 import PlayerCard from '../components/card/PlayerCard';
 import HostActionCard from '../components/card/HostActionCard';
 import { startGame } from '../api/Game';
 import {
   getRoom, Room, changeRoomHost, updateRoomSettings, removeUser,
 } from '../api/Room';
-import { NumberInput } from '../components/core/Input';
 import { errorHandler } from '../api/Error';
-import { ThemeConfig } from '../components/config/Theme';
+import {
+  CopyIndicator,
+  CopyIndicatorContainer,
+  InlineCopyIcon,
+  InlineBackgroundCopyText,
+} from '../components/special/CopyIndicator';
+import IdContainer from '../components/special/IdContainer';
+import { FlexBareContainer } from '../components/core/Container';
+import { Slider, SliderContainer } from '../components/core/RangeSlider';
 
 type LobbyPageLocation = {
   user: User,
   roomId: string,
 };
 
+const HeaderContainer = styled.div`
+  text-align: left;
+  width: 66%;
+  margin: 2rem auto 0;
+
+  @media (max-width: 750px) {
+    width: 80%;
+  }
+  @media (max-width: 600px) {
+    width: 100%;
+  }
+`;
+
+const PrimaryButtonZeroLeftMargin = styled(PrimaryButton)`
+  margin-left: 0;
+`;
+
+const FlexBareContainerLeft = styled(FlexBareContainer)`
+  text-align: left;
+`;
+
+const PlayersContainer = styled.div`
+  flex: 6;
+  padding-right: 5px;
+`;
+
+const RoomSettingsContainer = styled.div`
+  flex: 4;
+  padding-left: 5px;
+`;
+
+const SmallHeaderTextZeroTopMargin = styled(SmallHeaderText)`
+  margin: 0 0 10px 0;
+`;
```

Comment:
Slight personal preference but for some of these components, could we rename them to something more descriptive of its use case as opposed to the type of style? i.e. maybe something like SmallSettingsHeaderText for this, and maybe just StartGameButton instead of PrimaryButtonZeroLeftMargin for example (that way in the future if we ever change these styles we won't have to refactor the names of all the components to reflect the CSS)

Reply:
I think that makes sense, sounds good 👍 I had originally named it that because it was used in two different cases, but I simplified it to use the name `LobbyContainerTitle`.